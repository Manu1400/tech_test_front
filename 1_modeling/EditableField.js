import React, { Component } from 'react';

class EditableField extends Component {
    constructor(props) {
        super(props);
    
        this.state = {
          isReadOnly: true,
          text: this.props.value
        };
      
        this.handleChange = this.handleChange.bind(this);
    }

    validate() {
        this.setState({
            isReadOnly: true
        })
    }
  
    activateEditMode() {
        this.setState({
            isReadOnly: false
        })
    }
  
    handleChange(event) {
        this.setState({
            text: event.target.value
        })
    }
    
    render() {
        return <div>
          <label>{this.props.name}</label>
          <br />
            <input type='text' name='simple-input' 
                value={this.state.text}
                onChange={this.handleChange}
                onClick={() => this.activateEditMode()}
                readOnly={this.state.isReadOnly} />
            {!this.state.isReadOnly &&
                <input type="button" 
                    className="glyphicon glyphicon-pencil"
                    value="Validate"
                    onClick={() => this.validate()} />
            }
        </div>
    }
}

export default EditableField