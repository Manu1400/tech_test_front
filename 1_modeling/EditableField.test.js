import React from 'react';
import EditableField from './EditableField';
import renderer from 'react-test-renderer';

test('Display EditableField component', () => {
  const component = renderer.create(
    <EditableField name="Name" value="Charlie Chaplin" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

/* logical tests */
