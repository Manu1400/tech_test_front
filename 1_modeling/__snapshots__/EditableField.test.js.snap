// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial state 1`] = `
<div>
  <label>
    Name
  </label>
  <br />
  <input
    name="simple-input"
    onChange={[Function]}
    onClick={[Function]}
    readOnly={true}
    type="text"
    value="Charlie Chaplin"
  />
</div>
`;
